internal/storage/db/internal/db.go

```diff
@@ -478,7 +480,72 @@ func (s *dbStorage) GetDependents(ctx context.Context, ids ...namer.ModuleID) (m
 	return out, nil
 }
 
-func (s *dbStorage) Disable(ctx context.Context, policyKey ...string) (int64, error) {
+func (s *dbStorage) HasDescendants(ctx context.Context, ids ...namer.ModuleID) (map[namer.ModuleID]bool, error) {
+	// SELECT 1
+	// FROM policy_ancestor pa1 JOIN policy p1 ON (pa1.policy_id = p1.id AND p1.disabled = false)
+	// WHERE pa1.ancestor_id = p.id;
+	innerQuery := s.db.Select(goqu.L("1")).
+		From(goqu.T(PolicyAncestorTbl).As("pa1")).
+		Join(
+			goqu.T(PolicyTbl).As("p1"),
+			goqu.On(
+				goqu.And(
+					goqu.C(PolicyAncestorTblPolicyIDCol).Table("pa1").Eq(goqu.C(PolicyTblIDCol).Table("p1")),
+					goqu.C(PolicyTblDisabledCol).Table("p1").Eq(goqu.V(false)),
+				),
+			),
+		).
+		Where(goqu.C(PolicyAncestorTblAncestorIDCol).Table("pa1").Eq(goqu.C(PolicyTblIDCol).Table("p")))
+
+	// SELECT p.id, EXISTS(<innerQuery>) AS has_descendants
+	// FROM policy p
+	// WHERE p.id IN (?);
+	query := s.db.Select(
+		goqu.C(PolicyTblIDCol).Table("p"),
+		goqu.L("EXISTS ?", innerQuery).As("has_descendants"),
+	).
+		From(goqu.T(PolicyTbl).As("p")).
+		Where(goqu.C(PolicyTblIDCol).Table("p").In(ids))
+	result, err := query.Executor().ScannerContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not execute %q query: %w", "GetAncestorsAndDescendants", err)
+	}
+	defer result.Close()
+
+	out := make(map[namer.ModuleID]bool)
+	for result.Next() {
+		var rec struct {
+			ID             namer.ModuleID `db:"id"`
+			HasDescendants bool           `db:"has_descendants"`
+		}
+		if err := result.ScanStruct(&rec); err != nil {
+			return nil, err
+		}
+
+		out[rec.ID] = rec.HasDescendants
+	}
+
+	return out, nil
+}
+
+func (s *dbStorage) Disable(ctx context.Context, policyKey ...string) (uint32, error) {
+	var brokenChainPolicies []string
+	for _, pk := range policyKey {
+		mID := namer.GenModuleIDFromFQN(namer.FQNFromPolicyKey(pk))
+		hasDescendants, err := s.HasDescendants(ctx, mID)
```

Comment:
Instead of running the query for each policy to be disabled, build a list of  module IDs from the `policyKey` slice and call `HasDescendants` once with that list. Then iterate through the result of that call and print the error message for those policies that have descendants.